### PR TITLE
Migrate `AsyncDialogs` to Material 3

### PR DESCRIPTION
**Pull request by neeldoshii**
## Description
Currently the issue was that the colour of the checkbox was not matching with the APP theme `@color/material_light_blue_500`.  Initially, my thought process was to add a style in the Theme to ensure the checkbox color matches the app theme. However, I later realized that migrating to Material3 would be a better approach, as it allows the checkbox color to adapt dynamically based on the app theme. Since we plan to migrate to Material3 eventually, this seems like the most forward-thinking solution.

* 🌟 h/t to thank @SanjaySargam  for helping inspire this solution with #15184

* 🌟 h/t to thank @Thejas775 for helping inspire this solution with #16485

## Fixes
* Fixes : #15174
* Related : #13878

## Steps to Reproduce
1. App Info
2. Go to Manage Space
3. Click on Delete backups
4. Click on any one of the Checkbox Item

## Screenshot/Video

https://github.com/ankidroid/Anki-Android/assets/60827173/d72e6ef2-66be-48ab-a0a2-00378ed5e7dc

## Approach
In the PR :)

## Learning (optional, can help others)
https://github.com/material-components/material-components-android/blob/master/docs/components/Dialog.md

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by neeldoshii** [comment]
Just tested it on Android Version 9 which doesn't support Dynamic color.
Seems like it breaking on devices below android>12. Is there any way to handle this fallback mechanism to take theme color on device below 12

**Light Mode**
| Unselected (Light Mode) | Selected (Light Mode)|
|--------|--------|
| telegram-cloud-photo-size-5-6255924479045778269-y | telegram-cloud-photo-size-5-6255924479045778271-y |
| Unselected (Dark Mode) | Selected (Dark Mode)|
|telegram-cloud-photo-size-5-6255924479045778270-y|telegram-cloud-photo-size-5-6255924479045778272-y|

**Comment by BrayanDSO** [comment]
Seems like incomplete

**Comment by neeldoshii** [comment]
Created Styling to support for both API Level
| Android 13 | Android 8 (API 26) |
|--------|--------|
| screenshot | screenshot |
| screenshot | screenshot | 
| screenshot|screenshot|
| screenshot|screenshot|

**The above screenshots is of Delete Restore Backups following black,dark,plain, light theme**

**Android 13** (Plain, Light, Dark, Black)
| Delete Collection | Delete Everything |
|--------|--------|
|screenshot| screenshot |
| screenshot | screenshot |
| screenshot | screenshot |
| screenshot | screenshot |

**Android 8 (API 26)** (Plain, Light, Dark, Black)
| Delete Collection | Delete Everything |
|--------|--------|
| screenshot | screenshot |
| screenshot | screenshot |
| screenshot | screenshot |
| screenshot | screenshot |

File: AnkiDroid/src/main/res/values/theme_black.xml [hunk 1]
```diff
@@ -105,6 +105,9 @@
         <!-- AlertDialogTheme-->
         <item name="alertDialogTheme">@style/AlertDialogStyle</item>
 
+        <!-- MaterialAlertDialog-->
+        <item name="materialAlertDialogPrimaryColor">@color/theme_black_primary_dark</item>
```
Comment (neeldoshii) [inline AnkiDroid/src/main/res/values/theme_black.xml:109]: I am still confused how is this taking light grey color in the dialog where here I have passed `<color name="theme_black_primary_dark">#000000</color>`.  Not sure why its not overriding leading both `theme_black` and `theme_dark` somehow both of the theme has same color.

PS : This is my first time working into theming.

**Comment by BrayanDSO** [comment]
Already implemented

**Pull request closed without merging**